File: client/src/service/TimelineService.js
import axios from "axios";

export default{
    getTimeline(){
        return axios.get("/api/timeline");
    },

    getTimelineById(id){
        return axios.get(`/api/timeline/${id}`);
    },
    createTimeline(timeline){
        return axios.post("/api/timeline", timeline);
    },
    updateTimeline(id, timeline){
        return axios.put(`/api/timeline/${id}`, timeline);
    },
    deleteTimeline(id){
        return axios.delete(`/api/timeline/${id}`);
    }
}

File: server/src/main/java/com/usforever/service/TimelineService.java
package com.usforever.service;

import com.usforever.model.Timeline;

import java.util.List;

public interface TimelineService {
    List<Timeline> getAllTimeline();

    Timeline getTimelineById(int id);

    Timeline createTimeline (Timeline newTimeline);

    Timeline updateTimeline (Timeline updatedTimeline);

    int deleteTimeline(int id);
}
